compiler/ir/ir.psi2ir/src/org/jetbrains/kotlin/psi2ir/generators/LoopExpressionGenerator.kt

```diff
@@ -197,7 +197,7 @@ class LoopExpressionGenerator(statementGenerator: StatementGenerator) : Statemen
                     irNextCall
                 )
             } else {
-                scope.createTemporaryVariable(irNextCall, "loop_parameter", origin = IrDeclarationOrigin.FOR_LOOP_IMPLICIT_VARIABLE)
+                scope.createTemporaryVariable(irNextCall, "loop_parameter")
```

Comment:
@madsager Maybe we should switch K/N to common for-lowers in first cause FOR_LOOP_IMPLICIT_VARIABLE is used in K/N HeaderProcessor. But from opposite I don't see any usage of it outside of them so looks like there would be safe to perform it even now.

@homuroll Would it be a problem with serialized ir if we remove FOR_LOOP_IMPLICIT_VARIABLE is some feature?

Reply:
@max-kammerer https://github.com/JetBrains/kotlin-native/commit/e24062155083feef3838791f6e69cb72ceceadda

Reply:
@max-kammerer Switch to common for-loop lowering happened here: https://github.com/JetBrains/kotlin-native/pull/3294 :-)

There is still one occurrence of FOR_LOOP_IMPLICIT_VARIABLE in the bitcode generator which can be removed once this lands here and is pulled into kotlin-native. I therefore believe this is safe.